src/test/java/seedu/sudohr/logic/commands/employee/EditCommandTest.java

```diff
@@ -197,6 +204,70 @@ public void execute_invalidEmployeeIdFilteredList_failure() {
+
+        // check cascaded to department level, since employee is identified by id, old employee object
+        // should not be found in list whereas new employee object should be found.
+        assertTrue(!modelWithDepts.getDepartment(HUMAN_RESOURCES.getName()).hasEmployee(employeeToEdit));
+        assertTrue(modelWithDepts.getDepartment(HUMAN_RESOURCES.getName()).hasEmployee(editedEmployee));
+        // check all other fields remain the same
+        Employee extractFromDept = modelWithDepts
+                .getDepartment(HUMAN_RESOURCES.getName())
+                .getEmployee(new Id(VALID_ID_AMY));
+        assertTrue(extractFromDept.getName().equals(employeeToEdit.getName()));
+        assertTrue(extractFromDept.getPhone().equals(employeeToEdit.getPhone()));
+        assertTrue(extractFromDept.getEmail().equals(employeeToEdit.getEmail()));
+        assertTrue(extractFromDept.getTags().equals(employeeToEdit.getTags()));
+        assertTrue(extractFromDept.getAddress().equals(employeeToEdit.getAddress()));
+    }
+
+    @Test
+    public void execute_cascadeUpdateEmployeeFromTwoDepartments_success() {
+        Employee employeeToEdit = EMPLOYEE_IN_HUMAN_RESOURCES_AND_SALES;
+        assertTrue(employeeToEdit != null);
+        // ensure employee exists in HUMAN_RESOURCES and SALES
+        assertTrue(modelWithDepts.getDepartment(HUMAN_RESOURCES.getName()).hasEmployee(employeeToEdit));
+        assertTrue(modelWithDepts.getDepartment(SALES.getName()).hasEmployee(employeeToEdit));
+
+        Employee editedEmployee = new EmployeeBuilder(EMPLOYEE_IN_HUMAN_RESOURCES_AND_SALES).withEmail(VALID_EMAIL_AMY)
+                .build();
+        EditCommand.EditEmployeeDescriptor descriptor =
+                new EditEmployeeDescriptorBuilder(EMPLOYEE_IN_HUMAN_RESOURCES_AND_SALES).withEmail(VALID_EMAIL_AMY)
+                .build();
+        String expectedMessage = String.format(EditCommand.MESSAGE_EDIT_EMPLOYEE_SUCCESS, editedEmployee);
+        Model expectedModel = new ModelManager(modelWithDepts.getSudoHr(), new UserPrefs());
+        expectedModel.setEmployee(employeeToEdit, editedEmployee);
+
+        // check edit command
+        EditCommand editCommand = new EditCommand(employeeToEdit.getId(), descriptor);
+        assertCommandSuccess(editCommand, modelWithDepts, expectedMessage, expectedModel);
+
+        assertTrue(modelWithDepts.getDepartment(SALES.getName()).hasEmployee(editedEmployee));
+        assertTrue(modelWithDepts.getDepartment(HUMAN_RESOURCES.getName()).hasEmployee(editedEmployee));
+        // check employee name field in each department has been updated
+        assertTrue(modelWithDepts.getDepartment(SALES.getName())
+                .getEmployee(employeeToEdit.getId()).getEmail().equals(new Email(VALID_EMAIL_AMY)));
+        assertTrue(modelWithDepts.getDepartment(HUMAN_RESOURCES.getName())
+                .getEmployee(employeeToEdit.getId()).getEmail().equals(new Email(VALID_EMAIL_AMY)));
+    }
+
```

Comment:
and add cascade check for leaves over here as well, thanks!

Reply:
okie